grc/gui/NotebookPage.py

```diff
@@ -152,6 +153,30 @@ def set_proc(self, process):
         """
         self.process = process
 
+    def term_proc(self):
+        """
+        Terminate the subprocess object
+
+        Add a callback to kill the process
+        after 4 seconds if not already terminated
```

Comment:
Might not be at all important, but as you changed the time in the code, you might as well change it in the docstring.